### PR TITLE
Pickers: convert to CSS-in-JS Part 1

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: #5411
- [X] Include a change request file using `$ npm run change`

#### Description of changes

This PR brings the first part of the pickers' conversion to use CSS-in-JS. Things that were done in this part:

1. Add `BasePicker.styles.ts` to provide CSS-in-JS styling for the already existing pickers (TagPicker, PeoplePicker) and still leave support for Sass styles for custom pickers implemented by extending from `BasePicker` class without injecting the default styles.
2. Added comments on explaining why the above is needed.
3. Add `styles` and `theme` optional props to `IBasePickerProps` interface.
4. Changed tests in `BasePicker.test.ts` to use correctly a public method on the component instance by accessing its `componentRef` (this surfaced after wrapping the pickers with the `styled` function and the linter giving me red lines).

5. Removing this parameter in the `PeoplePicker`'s `createGenericItem` default prop added by #4376
https://github.com/OfficeDev/office-ui-fabric-react/blob/31783e67b172c645d15e169b54c84e8bfed5d6ca/packages/office-ui-fabric-react/src/components/pickers/PeoplePicker/PeoplePicker.tsx#L64
It was never actually working and for some reason, this was not caught by types because this prop is only allowing to pass 2 arguments:
https://github.com/OfficeDev/office-ui-fabric-react/blob/31783e67b172c645d15e169b54c84e8bfed5d6ca/packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts#L120
... and later is called only with those 2 arguments in here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/31783e67b172c645d15e169b54c84e8bfed5d6ca/packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx#L795
After wrapping the Base classes with the `styled` function I started to get type errors. So I discussed it with @joschect and I consider it's safe to be removed. If later this functionality needs to be added (it was never actually working in the first place) this should be done in a separate PR with more changes than the initial mentioned PR introduced. @cliffkoh would appreciate your input with this surface API change as this function was exported even though it's just a default prop provided by `PeoplePicker`. Even if someone did import to use it I don't see a way to pass that argument in it.
6. A little cleanup of the TagPicker example by moving a state interface into the same file as the example for user visibility and some more type fixes. Plus fixing the same issue as in tests file with the improper use of the public method of the component instance. This was causing the linter to complain that property `items` does not exist on `TagPicker`. Changing it to `TagPickerBase` was solving the errors but user should not care about the Base and use the `componentRef` prop instead.
7. Updated the API file and snapshots to reflect the current changes

Part 2 PR will consist of converting the picker items (Tags, Personas).

#### Focus areas to test

Need some feedback on moving the string classNames from being always on the component elements to the styles file, where in case globalClassNames are disabled in the Theme, might break some already existing custom Sass overrides in consumers apps. Should I skip the step of `getGlobalClassNames(GlobalClassNames, theme)`?

Also, regarding the number 6 in my list above, because we had this example displayed for a long time for users to see, now that accessing those methods is not working that way, we might get into breaking someone's app with this change in case they used this example. Should we still go with this change?

### CODE REVIEW

**Comment by Vitalius1** [comment]
Have no idea what are those screener changes mean. It doesn't look like they changed how they look.

**Comment by cliffkoh** [comment]
Their values changed, e.g.

image

File: packages/office-ui-fabric-react/src/components/pickers/TagPicker/TagPicker.tsx [hunk 1]
```diff
@@ -15,8 +16,8 @@ export interface ITag {
 
 export interface ITagPickerProps extends IBasePickerProps<ITag> {}
 
-export class TagPicker extends BasePicker<ITag, ITagPickerProps> {
-  protected static defaultProps = {
+export class TagPickerBase extends BasePicker<ITag, ITagPickerProps> {
+  public static defaultProps = {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/pickers/TagPicker/TagPicker.tsx]: rename this file to tagPicker.ts

File: packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx [hunk 2]
```diff
@@ -24,8 +23,13 @@ const _testTags = [
   'yellow'
 ].map(item => ({ key: item, name: item }));
 
+export interface ITagPickerDemoPageState {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx]: >export [](start = 0, length = 6)

nit: unnecessary export

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts [hunk 3]
```diff
@@ -21,6 +22,10 @@ export interface IBasePicker<T> {
 // and searched for by the people picker. For example, if the picker is
 // displaying persona's than type T could either be of Persona or Ipersona props
 export interface IBasePickerProps<T> extends React.Props<any> {
+  /**
+   * Optional callback to access the IBasePicker interface. Use this instead of ref for accessing
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: nice

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
 import { ISuggestionsProps } from './Suggestions/Suggestions.types';
 import { SuggestionsController } from './Suggestions/SuggestionsController';
-import { IBasePicker, IBasePickerProps, ValidationState } from './BasePicker.types';
+import { IBasePicker, IBasePickerProps, ValidationState, IBasePickerStyleProps, IBasePickerStyles } from './BasePicker.types';
 import { IAutofill, Autofill } from '../Autofill/index';
 import { IPickerItemProps } from './PickerItem.types';
 import { IPersonaProps } from '../Persona/Persona.types';
 import * as stylesImport from './BasePicker.scss';
-const styles: any = stylesImport;
+const legacyStyles: any = stylesImport;
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: >legacyStyles [](start = 6, length = 12)

nice

File: packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx (hunk 2 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx]: Resolved. I moved it from a separate file and forgot to remove the export)))

**Comment by cliffkoh** [comment]
@vitalius1 I don't see any issue with number 5 :) It seems sound to me.

**Pull request merged**